Gordon360/Services/HousingService.cs

```diff
@@ -321,234 +326,217 @@ public int EditApplication(string editorId, string sess_cde, int apartAppId, str
-                buildingCodeParam = new SqlParameter("@BLDG_CDE", bldg);
-                apartmentChoiceResult = RawSqlQuery<ApartmentChoiceSaveViewModel>.query("DELETE_AA_APARTMENT_CHOICES @APPLICATION_ID, @BLDG_CDE", appIdParam, buildingCodeParam); // run stored procedure
+                appIDParam = new SqlParameter("@APPLICATION_ID", applicationID);
+                rankingParam = new SqlParameter("@RANKING", apartmentChoice.HallRank);
+                buildingCodeParam = new SqlParameter("@BLDG_CDE", apartmentChoice.HallName);
+
+                apartmentChoiceResult = RawSqlQuery<AA_ApartmentChoices>.query("DELETE_AA_APARTMENT_CHOICE @APPLICATION_ID, @BLDG_CDE", appIDParam, buildingCodeParam); //run stored procedure
                 if (apartmentChoiceResult == null)
                 {
-                    throw new ResourceNotFoundException() { ExceptionMessage = "The apartment preference could not be saved." };
+                    throw new ResourceNotFoundException() { ExceptionMessage = "Apartment choice with ID " + applicationID + " and hall name " + apartmentChoice.HallName + " could not be removed." };
                 }
             }
 
             //--------
-            // Update the date modified
+            // Update the date modified (and application editor if necessary)
 
-            IEnumerable<ApartmentAppSaveResultViewModel> result = null;
+            IEnumerable<ApartmentApplicationViewModel> result = null;
 
             DateTime now = System.DateTime.Now;
 
-            appIdParam = new SqlParameter("@APPLICATION_ID", apartAppId);
+            appIDParam = new SqlParameter("@APPLICATION_ID", applicationID);
             SqlParameter timeParam = new SqlParameter("@NOW", now);
-
-            result = RawSqlQuery<ApartmentAppSaveResultViewModel>.query("UPDATE_AA_APPLICATION_DATEMODIFIED @APPLICATION_ID, @NOW", appIdParam, timeParam); //run stored procedure
-            if (result == null)
+            if (newEditorID != storedEditorID)
             {
-                throw new ResourceNotFoundException() { ExceptionMessage = "The application DateModified could not be updated." };
+                SqlParameter editorParam = new SqlParameter("@EDITOR_ID", userID);
+                SqlParameter newEditorParam = new SqlParameter("@NEW_EDITOR_ID", newEditorID);
+                result = RawSqlQuery<ApartmentApplicationViewModel>.query("UPDATE_AA_APPLICATION_EDITOR @APPLICATION_ID, @EDITOR_ID, @NOW, @NEW_EDITOR_ID", appIDParam, editorParam, timeParam, newEditorParam); //run stored procedure
+                if (result == null)
+                {
+                    throw new ResourceCreationException() { ExceptionMessage = "The application could not be updated." };
+                }
+            }
+            else
+            {
+                result = RawSqlQuery<ApartmentApplicationViewModel>.query("UPDATE_AA_APPLICATION_DATEMODIFIED @APPLICATION_ID, @NOW", appIDParam, timeParam); //run stored procedure
```

Comment:
Should this be done in every case, not just when the editor is not modified?

Reply:
dateModified is automatically updated when you update the editor.